plutus-tx/src/Language/PlutusTx/Evaluation.hs

```diff
@@ -0,0 +1,34 @@
+module Language.PlutusTx.Evaluation (evaluateCek, evaluateCekLog) where
+
+import           Language.PlutusCore.Interpreter.CekMachine hiding (evaluateCek)
+import           Language.PlutusCore
+import           Language.PlutusCore.Constant.Dynamic
+import           Language.PlutusCore.Constant
+
+import           Control.Exception
+
+import           System.IO.Unsafe
+
+stringBuiltins :: DynamicBuiltinNameMeanings
+stringBuiltins =
+    insertDynamicBuiltinNameDefinition dynamicCharToStringDefinition $ insertDynamicBuiltinNameDefinition dynamicAppendDefinition mempty
+
+-- TODO: pretty sure we shouldn't need the unsafePerformIOs here, we should expose a pure interface even if it has IO hacks under the hood
+
+-- | Evaluate a program in the CEK machine with the usual string dynamic builtins.
+evaluateCek
+    :: Program TyName Name ()
+    -> EvaluationResult
+evaluateCek p = unsafePerformIO $ evaluate $ runCek stringBuiltins p
+
+-- | Evaluate a program in the CEK machine with the usual string dynamic builtins and tracing, additionally
+-- returning the trace output.
+evaluateCekLog
+    :: Program TyName Name ()
+    -> ([String], EvaluationResult)
```

Comment:
We could keep this in the honest `IO` at the cost of complicating the golden test which I think is fine, but that's not too important I guess.

Reply:
It's not the test that I'm worried about, it's all the wallet-api stuff, which is very annoying if this stops being pure.